### ISSUE TITLE
Some lr-schedulers docs seems to have typos/missing information

### ISSUE THREAD

**Issue by Jakobhenningjensen** [pytorch/pytorch#68058]
## 📚 Documentation

According to [ConstantLR](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ConstantLR.html#torch.optim.lr_scheduler.ConstantLR) "*Decays the learning rate of each parameter group by a small constant factor until the number of epoch reaches a pre-defined milestone: total_iters*"  i.e with `factor=0.5`, `lr=0.05`  and `total_iters=4` I would assume

```
#lr = 0.025 
#lr = 0.0125
#lr = 0.00625
#lr = 0.003125
#lr = 0.5  #Back to initial
```
i.e

```
lr[i+1] = lr[0]*start_factor**i
```
where the docs writes 

```
 # Assuming optimizer uses lr = 0.05 for all groups
# lr = 0.025   if epoch == 0
# lr = 0.025   if epoch == 1
# lr = 0.025   if epoch == 2
# lr = 0.025   if epoch == 3
# lr = 0.05    if epoch >= 4
```

In the  [LinearLR](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.LinearLR.html) the learning-rate examples

```
# Assuming optimizer uses lr = 0.05 for all groups
# lr = 0.025    if epoch == 0
# lr = 0.03125  if epoch == 1
# lr = 0.0375   if epoch == 2
# lr = 0.04375  if epoch == 3
# lr = 0.005    if epoch >= 4
 scheduler = LinearLR(self.opt, start_factor=0.5, total_iters=4)
```

increases after epoch 0 until epoch 4 where it drops again (why?). As far as I understand LinearLR we get the learning-rate in each step by

```
lr[i+1] = lr[i]*start_factor**(i)
```

Generally it would be great, if you provided a (short) formular of how you calculate each learning rate in the schedulers since the wording can be interpreted differently by different persons but formulars (often) does not.

**Comment by albanD** [comment]
Hi,

Yes the LinearLR has a typo that was fixed in master: https://pytorch.org/docs/master/generated/torch.optim.lr_scheduler.LinearLR.html?highlight=linearlr#torch.optim.lr_scheduler.LinearLR

We can add a small formula there to improve the explanation, we would accept a PR adding that to the doc!

**Comment by Jakobhenningjensen** [comment]
I would love to provide a PR - the issue is, that I'm strugling to understand how the learning-rate is updated (due to the explanations above) thus providing a PR I'm not 100% certain of would a bit ...,

**Comment by albanD** [comment]
Yes the computation is a bit tricky.
It is mainly doing a linear interpolation between start_factor * lr at epoch 0 and end_factor * lr at epoch total_iters. I'm not sure if there is a simple formula to convey that?

**Comment by jaketae** [comment]
Given `lr`, the learning rate of the optimizer, and `delta = lr * (end_factor - start_factor) / total_iters`, I think we can write the formula (in code) as `lr[i] = start_factor * lr + delta * I`.

Would this be helpful to have in the docs?

**Comment by albanD** [comment]
That looks mostly ok. I think the `min(i, total_iters)` is important here as well.
But you can send a PR with that for sure!

**Comment by ozancaglayan** [comment]
and also the equation for LinearLR, I suspect isn't merged @jaketae right? Do you want to follow up this?

**Comment by jaketae** [comment]
@ozancaglayan Sorry for the late reply. Yes, I think I was carried away by other things and ended up not being able to wrap things up :( Feel free to pick up where I left off, if that'd be of any help!
